sdk/core/Azure.Core/src/Shared/Utf8JsonReaderExtensions.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using Azure.Core.Serialization;
+
+namespace Azure.Core
+{
+    internal static class Utf8JsonReaderExtensions
+    {
+        public delegate void PropertySetter<T>(ReadOnlySpan<byte> propertyName, ref T properties, ref Utf8JsonReader reader, ModelSerializerOptions options);
+        private delegate object ReadValue(ref Utf8JsonReader reader);
+
+        private static readonly Dictionary<Type, ReadValue> _readerActions = new Dictionary<Type, ReadValue>()
+        {
+            { typeof(string), (ref Utf8JsonReader reader) => reader.GetString() },
+            { typeof(bool), (ref Utf8JsonReader reader) => reader.GetBoolean() },
+            { typeof(int), (ref Utf8JsonReader reader) => reader.GetInt32() },
+            { typeof(long), (ref Utf8JsonReader reader) => reader.GetInt64() },
+            { typeof(double), (ref Utf8JsonReader reader) => reader.GetDouble() },
+            { typeof(float), (ref Utf8JsonReader reader) => reader.GetSingle() },
+            { typeof(decimal), (ref Utf8JsonReader reader) => reader.GetDecimal() },
+            { typeof(DateTimeOffset), (ref Utf8JsonReader reader) => reader.GetDateTimeOffset() },
+            { typeof(Guid), (ref Utf8JsonReader reader) => reader.GetGuid() },
+            { typeof(byte[]), (ref Utf8JsonReader reader) => reader.GetBytesFromBase64() },
+        };
+
+        private static readonly Dictionary<Type, Func<string, object>> _keyConverter = new Dictionary<Type, Func<string, object>>()
+        {
+            { typeof(int), str => Convert.ToInt32(str) },
+            { typeof(string), str => str },
+        };
+
+        private static string ConvertKeyToString(Type typeToConvert, string key) => key;
+
+        public static List<T> GetList<T>(this ref Utf8JsonReader reader, ModelSerializerOptions options)
+        {
+            List<T> list = new List<T>();
+
+            if (reader.TokenType == JsonTokenType.Null)
```

Comment:
Seeing more of these without curly braces, you should get blue squiggly lines based on .editorconfig in our repo root. If you press Ctrl+. on one of those, you can fix all occurrences in a project.